src/astro/databases/google/bigquery.py

```diff
@@ -135,3 +150,96 @@ def merge_table(
             )
             statement += f" WHEN MATCHED THEN {update_statement}"
         self.run_sql(sql_statement=statement)
+
+    def check_optimised_path(self, source_file: File, target_table: Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        file_type = self.OPTIMIZED_PATHS_SUPPORTED_FILE_TYPES.get(source_file.type.name)
+        location_type = self.OPTIMIZED_PATHS.get(source_file.location.location_type)
+        return bool(location_type and file_type)
+
+    def optimised_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+        """
+        method_name = self.OPTIMIZED_PATHS.get(source_file.location.location_type)
+        if method_name:
```

Comment:
If a file type isn't supported, it will silently pass here. Should we raise an error, what are your thoughts?

Reply:
Make sense, added code change.